lambdas/data-migration2/tests/test-pdrs.js

```diff
@@ -0,0 +1,336 @@
+const { dynamodbDocClient } = require('@cumulus/aws-client/services');
+const { getKnexClient, localStackConnectionEnv } = require('@cumulus/db');
+const { fakeCollectionFactory, fakeProviderFactory } = require('@cumulus/api/lib/testUtils');
+
+// eslint-disable-next-line node/no-unpublished-require
+const { migrationDir } = require('../../db-migration');
+const { migratePdrRecord, migratePdrs } = require('../dist/lambda/pdrs');
+const { RecordAlreadyMigrated } = require('../dist/lambda/errors');
+
+const migrateFakeCollectionRecord = async (record, knex) => {
+  const updatedRecord = translateApiCollectionToPostgresCollection(record);
+  const [id] = await knex(tableNames.collections).insert(updatedRecord).returning('cumulus_id');
+  return id;
+};
+
+const fakeEncryptFunction = async () => 'fakeEncryptedString';
+
+const migrateFakeProviderRecord = async (record, knex) => {
+  const updatedRecord = await translateApiProviderToPostgresProvider(record, fakeEncryptFunction);
+  const [id] = await knex(tableNames.providers).insert(updatedRecord).returning('cumulus_id');
+  return id;
+};
+
+const generateTestPdr = (collectionId, provider, executionId) => ({
+  pdrName: cryptoRandomString({ length: 5 }),
+  collectionId: collectionId,
+  provider: provider,
+  status: 'running',
+  progress: 10,
+  execution: executionId,
+  PANSent: false,
+  PANmessage: 'message',
+  stats: { total: 1, completed: 0, failed: 0, processing: 1 },
+  address: cryptoRandomString({ length: 5 }),
+  originalUrl: cryptoRandomString({ length: 5 }),
+  timestamp: Date.now(),
+  duration: 10,
+  createdAt: Date.now(),
+  updatedAt: Date.now(),
+});
+
+let collectionsModel;
+let providersModel;
+let pdrsModel;
+
+const testDbName = `data_migration_2_${cryptoRandomString({ length: 10 })}`;
+const testDbUser = 'postgres';
+
+test.before(async (t) => {
+  process.env.stackName = cryptoRandomString({ length: 10 });
+  process.env.system_bucket = cryptoRandomString({ length: 10 });
+
+  process.env.CollectionsTable = cryptoRandomString({ length: 10 });
+  process.env.ProvidersTable = cryptoRandomString({ length: 10 });
+  process.env.PdrsTable = cryptoRandomString({ length: 10 });
+
+  await createBucket(process.env.system_bucket);
+
+  collectionsModel = new Collection();
+  await collectionsModel.createTable();
+
+  providersModel = new Provider();
+  await providersModel.createTable();
+
+  pdrsModel = new Pdr();
+  await pdrsModel.createTable();
+
+  t.context.knexAdmin = await getKnexClient({
+    env: {
+      ...localStackConnectionEnv,
+      migrationDir,
+    },
+  });
+  await t.context.knexAdmin.raw(`create database "${testDbName}";`);
+  await t.context.knexAdmin.raw(`grant all privileges on database "${testDbName}" to "${testDbUser}"`);
+
+  t.context.knex = await getKnexClient({
+    env: {
+      ...localStackConnectionEnv,
+      PG_DATABASE: testDbName,
+      migrationDir,
+    },
+  });
+
+  await t.context.knex.migrate.latest();
```

Comment:
All of this boilerplate can be removed by using `generateLocalTestDb`. See https://github.com/nasa/cumulus/blob/feature-rds/packages/db/tests/models/test-granule-model.js#L20